bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/server/enable/HasLanguageServerPropertyTester.java

```diff
@@ -54,7 +54,7 @@ public boolean test(Object receiver, String property, Object[] args, Object expe
 				var isEnabled = LspUtils.getProject(uri).map(cLanguageServerProvider::isEnabledFor)
 						.orElse(LspUtils.isFileOpenedInLspEditor(uri));
 				if (isEnabled) {
-					initialFileManager.setInitialUri(uri);
+					ServiceCaller.callOnce(getClass(), InitialUri.class, iu -> iu.register(uri));
```

Comment:
> if there are performance drawbacks during runtime?

Are you aware of any performance drawbacks regarding `ServiceCaller#callOnce` @laeubi ?

Reply:
I would avoid `ServiceCaller` whenever possible and instead use either OSGi-DS (what seems to be used already here), White-Board or Extender pattern or similar. Beside that of course everything has a performance penalty but I won't expect it is noticeable here, while decoupling often makes the code easier to read and maintain.

Reply:
How would you recommend to access OSGi service from PropertyTester, that is instantiated from extension @laeubi ?
I thought `ServiceCaller` is the recommended way here to avoid low-level API.

Reply:
You can change the code that uses the extension to additionally accept services registered, that way you can make this a plain service with an `@Reference`

Reply:
I'm not sure if this approach is currently applicable for extenders of `org.eclipse.core.expressions.PropertyTester`

Reply:
Ah I thought it was an cdt extension :-\
I fear then there are not much options, just if you want to call the service more often you might want to save a service caller in a field so the service is not fetched and released each time, it just looks a bit strange for a property tester ti have side-effects like `addUrl(....)`

Reply:
> you might want to save a service caller in a field so the service is not fetched and released each time

thank you for advice. my concern was that for "cached" service caller I don't have proper place to call `unget` 

> it just looks a bit strange for a property tester ti have side-effects like addUrl(....)

Yes, this looks a bit terrible, we will think about it further to find better place

Reply:
>> you might want to save a service caller in a field so the service is not fetched and released each time

> thank you for advice. my concern was that for "cached" service caller I don't have proper place to call `unget`

I would prefer a cached service caller.  Can `ungetService` be called when the bundle gets stopped?

Reply:
If the bundle is stopped all service are automatically unget, I also don't see much advantage her in ungetting the service as it seems the component will stay used over the full life-time of the bundle anyways.